### DISCUSSION TITLE
When using grep_searcher crate

### DISCUSSION THREAD

**Discussion by bq-wrongway** [BurntSushi/ripgrep#2859] [Q&A] [answered]
I would like to have something similar to the example on the repo,
I created a matcher, (newline matcher)
I created searcher with search builder.

And later i use searcher.search_path()

My issue is that search path requires sink (printer) as a third param, i would like to add my search results to a vector or a map.
So i tried using example from https://docs.rs/grep-searcher/0.1.13/grep_searcher/index.html
And instead of using printer sink(standardstream),i used UTF8, however this does not work and i get no matches.

So basically,is there an option of using search_path() and saving results from it to a vector of strings?

**Comment by BurntSushi** [comment]
> So basically,is there an option of using search_path() and saving results from it to a vector of strings?

Yes, that's exactly what this example is doing: https://docs.rs/grep-searcher/0.1.13/grep_searcher/index.html#example

> however this does not work

[Help me help you.](https://www.youtube.com/watch?v=l1B1_jQnlFk) Don't tell, [_show_](https://en.wikipedia.org/wiki/Minimal_reproducible_example). Here's [how to do it](https://stackoverflow.com/help/minimal-reproducible-example).

**Reply by bq-wrongway** [reply]
Thank you so much for your reply, i wrote question on mobile and did not provide any code samples, i am sorry about that.
Here is pastebin with two functions, mind you its just bare bones. 
https://pastebin.com/bPrPqsqm

so the function that should add to vector, does not work at the moment, it just throws an error (i know its because i am using unwrap).
But i do try to handle the error i get same issue

/home/melnibone/Documents/trst/test.xt: line numbers not enabled 

i am not sure what is happening. 
Also i feel like i am doing something wrong by using matcher.find() inside of search_path() function, its like i am nesting search inside of search.

Thank you for your help, if needed i can make a whole example, but i think the function itself is enough to ilustrate what i am tring to accomplish

**Reply by BurntSushi** [reply, marked as answer]
Firstly, that's not an MRE. Please read the links I shared with you.

Secondly, you _aren't_ using the example in the docs, you're using something _different_.

Thirdly, the error message tells you that line numbers aren't enabled. Yet you have explicitly disabled line numbers. That seems like your problem...

**Reply by bq-wrongway** [reply]
Sorry, 
here is link to small repo https://github.com/bq-wrongway/grep_example 
I commented the functions, you are right i did made couple of dumb mistakes, like disabling line numbers. 
Mostly last question is part of comment in the 2nd function, i am using search_path(), and then inside of it i am using find() method, it feels like i am doing search 2 times.

First functions just simply prints out to std, i am not sure if find() is needed for the second time.

**Reply by BurntSushi** [reply]
You still haven't provided an MRE. I don't see any commands telling me what to run. Or any inputs. Or the output you're seeing. Or the expected/desired output.

I can't keep going back and forth like this. Please have someone help you through the process of creating an MRE.

**Reply by bq-wrongway** [reply]
Hi,
I apologize for creating confusion, my first issue was that i could not make the example work, with your help i realised what i am doing wrong and i fixed the issue, so my original complain is not valid anymore.

I linked repo that works, just running cargo run , will execute main method that has a call to function, that will search attached file with test query, but lets ignore that. What i needed i have achieved and it works.

What i dont understand is following :
code taken from your example :
https://github.com/BurntSushi/ripgrep/blob/master/crates/grep/examples/simplegrep.rs

Relevant part :

`let result = searcher.search_path(
                &matcher,
                dent.path(),
                printer.sink_with_path(&matcher, dent.path()),
            );`
            
            When i take a look at this, search_path gets a needle (matcher refrence) and it searches for it in path (haystack). and it just prints out to stdout.
            This part i understand, great!
            
           Code taken from your example here https://docs.rs/grep-searcher/0.1.13/grep_searcher/index.html 
           `Searcher::new().search_slice(&matcher, SHERLOCK, UTF8(|lnum, line| {
    // We are guaranteed to find a match, so the unwrap is OK.
    let mymatch = matcher.find(line.as_bytes())?.unwrap();
    matches.push((lnum, line[mymatch].to_string()));
    Ok(true)
}))?;`

so this looks very very much like first function, but here is what i dont understand. You pass &matcher to it, which i expect should be term that is being searched, then you pass what to search SHERLOCK slice, but then inside of UTF8 closure, you again use matcher.find().

This is what i dont understand, why do we pass &matcher, and then again we call matcher.find() inside of closure.
Does this mean that i am repeating search ? This is my actual question.

Thank you in advance, and i apologize for being a timesink for you
